.circleci/set-up-globals.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+set -ex
+
+#==================================================
+# The section would be transferable to a DOCKERFILE
+#==================================================
+
+# Update current apt packages
+apt-get update
+
+#=========================================================================
+# Commands below this line would not be transferable to a docker container
+#=========================================================================
+
+# Enable Composer parallel downloads
+composer global require -n "hirak/prestissimo:^0.3"
+
+# Install Terminus
+if [ ! -d $HOME/terminus ]
+then
+	# Clone terminus if it doesn't exist
+	echo -e "Installing Terminus...\n"
+	git clone --branch master git://github.com/pantheon-systems/terminus.git $HOME/terminus
+	cd $HOME/terminus
+	composer install
+	cd -
+else
+	# Otherwise make sure terminus is up to date
+	cd $HOME/terminus
+	git pull
+	composer install
+	cd -
+fi
+
+
+#=====================================================================================================================
+# Start EXPORTing needed environment variables
+# Circle CI 2.0 does not yet expand environment variables so they have to be manually EXPORTed
+# Once environment variables can be expanded this section can be removed
+# See: https://discuss.circleci.com/t/unclear-how-to-work-with-user-variables-circleci-provided-env-variables/12810/11
+# See: https://discuss.circleci.com/t/environment-variable-expansion-in-working-directory/11322
+# See: https://discuss.circleci.com/t/circle-2-0-global-environment-variables/8681
+#=====================================================================================================================
+echo 'export PATH=$PATH:$HOME/bin:$HOME/terminus/bin' >> $BASH_ENV
+echo 'export BRANCH=$(echo $CIRCLE_BRANCH | grep -v '"'"'^\(master\|[0-9]\+.x\)$'"'"')' >> $BASH_ENV
+echo 'export PR_ENV=${BRANCH:+pr-$BRANCH}' >> $BASH_ENV
+echo 'export CIRCLE_ENV=ci-$CIRCLE_BUILD_NUM' >> $BASH_ENV
+# If we are on a pull request
+if [[ $CIRCLE_BRANCH != "master" && -n ${CIRCLE_PULL_REQUEST+x} ]]
+then
+	# Then use a pr- branch/multidev
+	PR_NUMBER=${CIRCLE_PULL_REQUEST##*/}
+	PR_BRANCH="pr-${PR_NUMBER}"
+	echo "export DEFAULT_ENV=pr-${PR_NUMBER}" >> $BASH_ENV
+else
+	# otherwise make the branch name multidev friendly
+	if [[ $CIRCLE_BRANCH == "master" ]]
+	then
+		echo "export DEFAULT_ENV=dev" >> $BASH_ENV
+	else
+		echo 'export DEFAULT_ENV=$(echo ${PR_ENV:-$CIRCLE_ENV} | tr '"'"'[:upper:]'"'"' '"'"'[:lower:]'"'"' | sed '"'"'s/[^0-9a-z-]//g'"'"' | cut -c -11 | sed '"'"'s/-$//'"'"')' >> $BASH_ENV
+	fi
+fi
+echo 'export TERMINUS_ENV=${TERMINUS_ENV:-$DEFAULT_ENV}' >> $BASH_ENV
+source $BASH_ENV
```

Comment:
@greg-1-anderson removing the sourcing of `$BASH_ENV` and just using `export` directly causes these variables to not be passed to the next step.

See [this failed workflow](https://circleci.com/workflow-run/aeab6dd2-146a-4853-84b6-74977181ed46) vs. the [passing workflow](https://circleci.com/workflow-run/f99fa317-c811-4a4d-b285-cb96776d908c) after I went back to sourcing `$BASH_ENV`.

My guess is that when we source the `$BASH_ENV` file it makes them available globally to the entire job vs. a standard `export` only being available to the current step.